File: 1980-find-unique-binary-string/1980-find-unique-binary-string.py
class Solution:
    def findDifferentBinaryString(self, nums: List[str]) -> str:
        st = {int(x, 2) for x in nums}
        n = len(nums)
        for num in range(0, n + 1):
            if num not in st:
                ans = format(num, '016b')
                return ans[16-n:]





# class Solution:
#     def findDifferentBinaryString(self, nums: List[str]) -> str:
#         n = len(nums)
#         ans = ""
        
#         for i in range(n):
#             ch = nums[i][i]
#             ans += '1' if ch == '0' else '0'
        
#         return ans





# # class Solution:
# #     def __init__(self):
# #         self.st = set()
# #         self.ans = ""

# #     def solve(self, curr, n):
# #         if len(curr) == n:
# #             if curr not in self.st:
# #                 self.ans = curr
# #             return
# #         if self.ans:
# #             return

# #         self.solve(curr + '0', n)
# #         self.solve(curr + '1', n)

# #     def findDifferentBinaryString(self, nums: List[str]) -> str:
# #         self.st = set(nums)
# #         self.solve("", len(nums[0]))
# #         return self.ans

File: 1980-find-unique-binary-string/1980-find-unique-binary-string.java
class Solution {
    public String findDifferentBinaryString(String[] nums) {
        int n = nums.length;
        StringBuilder ans = new StringBuilder();
        
        for (int i = 0; i < n; i++) {
            char ch = nums[i].charAt(i);
            ans.append(ch == '0' ? '1' : '0');
        }
        
        return ans.toString();
    }
}







// class Solution {
//     Set<String> st = new HashSet<>();
//     String ans = "";

//     void solve(StringBuilder curr, int n) {
//         if (curr.length() == n) {
//             if (!st.contains(curr.toString())) {
//                 ans = curr.toString();
//             }
//             return;
//         }
//         if (!ans.isEmpty()) return;

//         curr.append('0');
//         solve(curr, n);
//         curr.deleteCharAt(curr.length() - 1);

//         curr.append('1');
//         solve(curr, n);
//         curr.deleteCharAt(curr.length() - 1);
//     }

//     public String findDifferentBinaryString(String[] nums) {
//         int n = nums[0].length();
//         for (String num : nums) st.add(num);
//         solve(new StringBuilder(), n);
//         return ans;
//     }
// }